packages/theme-check-common/src/checks/unrecognized-render-snippet-params/index.ts

```diff
@@ -69,9 +69,42 @@ export const UnrecognizedRenderSnippetParams: LiquidCheckDefinition = {
       }
     }
 
+    function reportUnknownAliases(
+      node: RenderMarkup,
+      liquidDocParameters: Map<string, LiquidDocParameter>,
+      snippetName: string,
+    ) {
+      if (node.alias && !liquidDocParameters.has(node.alias) && node.variable) {
+        const asAliasMatch = node.source.match(new RegExp(`as\\s+${node.alias}`));
+
+        const suggest = asAliasMatch
+          ? [
+              {
+                message: `Remove '${node.alias}'`,
+                fix: (fixer: any) => {
+                  if (node.variable) {
+                    return fixer.remove(
+                      node.variable.position.start,
+                      node.source.indexOf(asAliasMatch[0]) + asAliasMatch[0].length,
```

Comment:
```suggestion
                      node.source.slice(node.position.start).indexOf(asAliasMatch[0]) + asAliasMatch[0].length,
```

This is a bit dangerous. I had two render tags with `as foo` in it. `node.source` returns the entire file, so doing a `indexOf` could find the wrong one.